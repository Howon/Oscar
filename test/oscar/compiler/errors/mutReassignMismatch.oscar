message testNow()


actor Tester() {
  mut int x = 1;

  receive = {
    | testNow() => {
        x = 2.0;
      }
    | die() => {
      }
  }
}

def main() => unit = {
  actor<Tester> tester = spawn actor<Tester>();
  message<testNow>() |> tester;
  message<die>() |> tester;
}
