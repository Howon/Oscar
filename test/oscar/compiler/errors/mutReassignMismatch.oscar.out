Assignment to incompatible type: double cannot be assigned to int